src/lib/autocomplete/autocomplete.spec.ts

```diff
@@ -2607,3 +2639,19 @@ class AutocompleteWithNativeAutocompleteAttribute {
 })
 class InputWithoutAutocompleteAndDisabled {
 }
+
+@Component({
+  template: `
+    <input type="number" matInput [matAutocomplete]="selected ? auto1 : auto0">
+    <mat-autocomplete #auto0="matAutocomplete">
+      <mat-option [value]="0">First</mat-option>
+    </mat-autocomplete>
+
+    <mat-autocomplete #auto1="matAutocomplete">
+      <mat-option [value]="1">Second</mat-option>
+    </mat-autocomplete>
+  `,
+})
+class DynamicallyChangingAutocomplete {
+  selected = 0;
```

Comment:
This seems like a weird way to toggle between the panels. What you can do instead is to get a reference to both of them through a `viewChild` and assign them directly as properties.

Reply:
fixed